docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,272 @@
+
+=== Ladder
+
+* **Contributors**. Anyone can participate.
+  There is no precondition except having a GitHub account, just submit pull requests or comment in existing ones!
+* **Core Pull Request Reviewers** can be nominated by contributors in the link:https://groups.google.com/d/forum/jenkinsci-dev[developer mailing list]. 
+Self-nomination is fine.
+Decision is made by a consensus in the mailing list or via voting at the link:https://jenkins.io/project/governance-meeting/[governance meeting].
+* **Core Maintainers** - same process as for Core PR reviewers.
+  All nominees must also sign an link:https://github.com/jenkinsci/infra-cla/[Individual Contributor License Agreement] before getting the permission in GitHub repositories.
+* **Release Team Members** are assigned by the Jenkins Release officer
+
+== Pull request review process
+
+Jenkins core is a mission-critical part of the ecosystem.
+We need to ensure that submitted pull requests are not only code complete,
+but also that they do not introduce undesired defects, breaking changes and technical debt.
+At the same time, we are interested to make the review process as simple as possible for contributors and maintainers.
+
+=== Review goals
+
+Pull requests review in Jenkins is not just about reviewing code and accepting them if the code is OK.
+Core maintainers are expected to ensure feasibility and compatibility of changes,
+to maintain good quality of the codebase and documentation,
+and to ensure there is a consensus between contributors.
+
+==== Verifying Feasibility
+
+Reviewers are expected to look at changes critically from a "product management" point of view.
+It's not just about the code, but also whether the change makes sense in a global / holistic way, considering existing popular plugins and the way users experience Jenkins overall.
+Also, not every change needs to be merged into the core.
+Many features would be better introduced as plugins which have separate release cycles and allow delivering changes faster.
+We want to extend the Jenkins core and incorporate widely used functionality and extension points there,
+but we try to keep the core as minimal as possible in terms of dependencies.
+
+==== Ensuring Compatibility
+
+The Jenkins project has a long history of backward compatibility.
+We accept breaking changes when it is needed (security fixes, feature deprecation and removal, etc.),
+but ultimately the project's goal is retain as much compatibility as possible.
+It includes both feature compatibility and binary/API compatibility which is important for the plugin ecosystem.
+Although we have some tools (like https://github.com/jenkins-infra/usage-in-plugins[usage-in-plugins]) for checking API usages across open-source plugins,
+there is no way to confirm external usages in 3rd-party proprietary plugins which are also a part of the ecosystem.
```

Comment:
I do not want to rathole on this, but do we consider 3rd-party plugin to be part of the ecosystem? To me, it would make more sense to define the boundaries of the ecosystem (pretty much https://github.com/{jenkinsci,jenkins-infra,stapler}) and that would be the place where it is possible to make certain effort to ensure compatibility, but there is not commitment elsewhere.

Reply:
I consider 3rd-party plugins as a part of wider Jenkins ecosystem which we would like to respect and minimize breaking changes there. Not just because I work for CloudBees, there we can easily fix breaking changes. But there are thousands of in-house plugins created by Jenkins users. I have written at least one at all my previous companies